src/test/fuzz/vecdeque.cpp

```diff
@@ -0,0 +1,488 @@
+        }
+        // Clear out the buffers so we can check below that no objects exist anymore.
+        sim[i].clear();
+        real[i].clear();
+    }
+
+    if constexpr (CheckNoneLeft) {
+        tmp = std::nullopt;
+        T::CheckNoneExist();
+    }
+}
+
+/** Data structure with built-in tracking of all existing objects. */
+template<size_t Size>
+class TrackedObj
+{
+    static_assert(Size > 0);
+
+    /* Data type for map that actually stores the object data.
+     *
+     * The key is a pointer to the TrackedObj, the value is the uint64_t it was initialized with.
+     * Default-constructed and moved-from objects hold an std::nullopt.
+     */
+    using track_map_type = std::map<const TrackedObj<Size>*, std::optional<uint64_t>>;
+
+private:
+
+    /** Actual map. */
+    static inline track_map_type g_tracker;
+
+    /** Iterators into the tracker map for this object.
+     *
+     * This is an array of size Size, all holding the same value, to give the object configurable
+     * size. The value is g_tracker.end() if this object is not fully initialized. */
+    typename track_map_type::iterator m_track_entry[Size];
+
+    void Check() const
+    {
+        auto it = g_tracker.find(this);
+        for (size_t i = 0; i < Size; ++i) {
+            assert(m_track_entry[i] == it);
+        }
+    }
+
+    /** Create entry for this object in g_tracker and populate m_track_entry. */
+    void Register()
+    {
+        auto [it, inserted] = g_tracker.emplace(this, std::nullopt);
+        assert(inserted);
+        for (size_t i = 0; i < Size; ++i) {
+            m_track_entry[i] = it;
+        }
+    }
+
+    void Deregister()
+    {
+        Check();
+        assert(m_track_entry[0] != g_tracker.end());
+        g_tracker.erase(m_track_entry[0]);
+        for (size_t i = 0; i < Size; ++i) {
+            m_track_entry[i] = g_tracker.end();
+        }
+    }
+
+    /** Get value corresponding to this object in g_tracker. */
+    std::optional<uint64_t>& Deref()
+    {
+        Check();
+        assert(m_track_entry[0] != g_tracker.end());
+        return m_track_entry[0]->second;
+    }
+
+    /** Get value corresponding to this object in g_tracker. */
+    const std::optional<uint64_t>& Deref() const
+    {
+        Check();
+        assert(m_track_entry[0] != g_tracker.end());
+        return m_track_entry[0]->second;
+    }
+
+public:
+    ~TrackedObj() { Deregister(); }
+    TrackedObj() { Register(); }
+
+    TrackedObj(uint64_t value)
+    {
+        Register();
+        Deref() = value;
+    }
+
+    TrackedObj(const TrackedObj& other)
+    {
+        Register();
+        Deref() = other.Deref();
+    }
+
+    TrackedObj(TrackedObj&& other)
+    {
+        Register();
+        Deref() = other.Deref();
+        other.Deref() = std::nullopt;
+    }
+
+    TrackedObj& operator=(const TrackedObj& other)
+    {
+        Deref() = other.Deref();
+        return *this;
+    }
```

Comment:
This copy assignment can be amended to handle the self-assignment as well to do not conflict with https://github.com/bitcoin/bitcoin/pull/30234.